File: backend/models/schema.py
from pydantic import BaseModel, Field

class Reference(BaseModel):
    table: str = Field(description="Name of the referenced table")
    column: str = Field(description="Name of the referenced column")

class Column(BaseModel):
    name: str = Field(description="Name of the column")
    type: str = Field(description="SQL data type")
    constraints: list[str] = Field(default_factory=list, description="SQL constraints")
    references: Reference | None = Field(default=None, description="Foreign key reference, if this column is a foreign key")

class TableSchema(BaseModel):
    table_name: str = Field(description="Name of the table")
    columns: list[Column] = Field(description="Columns defined in the table")
    seed_inserts: list[str] = Field(default_factory=list, description="Seed INSERT SQL statements for this table")
    create_table_sql: str | None = Field(default=None, description="Generated CREATE TABLE SQL statement")
    inserts_sql: str | None = Field(default=None, description="Generated seed INSERT statements combined")

class SQLSchemaResponse(BaseModel):
    tables: list[TableSchema] = Field(description="Tables generated from user description")

class GenerateRequest(BaseModel):
    description: str = Field(description="Natural language description of the requested database schema")

File: frontend/src/types/schema.ts
export interface Reference {
  table: string;
  column: string;
}

export interface Column {
  name: string;
  type: string;
  constraints: string[];
  references: Reference | null;
}

export interface TableSchema {
  table_name: string;
  columns: Column[];
  seed_inserts: string[];
  create_table_sql: string | null;
  inserts_sql: string | null;
}

export interface SQLSchemaResponse {
  tables: TableSchema[];
}

export type Dialect = "postgres" | "mysql" | "sqlite";

export interface HistoryEntry {
  id: number;
  prompt: string;
  response: SQLSchemaResponse;
  dialect: Dialect;
}

export const DIALECTS: { value: Dialect; label: string }[] = [
  { value: "postgres", label: "PostgreSQL" },
  { value: "mysql", label: "MySQL" },
  { value: "sqlite", label: "SQLite" },
];
